nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/RemoveProxyAction.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ */
+
+package org.nuxeo.ecm.core.bulk.actions;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentModelList;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.lib.stream.computation.Computation;
+
+/**
+ * @since 10.3
+ */
+public class RemoveProxyAction extends AbstractBulkAction {
+
+    public static final String ACTION_NAME = "removeProxy";
+
+    public RemoveProxyAction() {
+        super(ACTION_NAME);
+    }
+
+    @Override
+    protected Computation createComputation(int batchSize, int batchThresholdMs) {
+        return new RemoveProxyComputation(getActionName(), batchSize, batchThresholdMs);
+    }
+
+    public static class RemoveProxyComputation extends AbstractBulkComputation {
+
+        public RemoveProxyComputation(String name, int batchSize, int batchThresholdMs) {
+            super(name, 1, 1, batchSize, batchThresholdMs);
+        }
+
+        @Override
+        protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            Collection<DocumentRef> refs = ids.stream().map(IdRef::new).collect(Collectors.toList());
+            DocumentModelList docs = session.getDocuments(refs.toArray(new DocumentRef[refs.size()]));
+            Collection<DocumentRef> proxies = docs.stream()
+                                                  .filter(DocumentModel::isProxy)
+                                                  .map(DocumentModel::getRef)
+                                                  .collect(Collectors.toList());
+            session.removeDocuments(proxies.toArray(new DocumentRef[0]));
+            session.save();
+        }
+
+    }
+}
```

Comment:
Maybe not for this one, but we should start putting actions in their associated module (here `nuxeo-core`) IMO, and not list all actions in the `nuxeo-core-bulk` module

Reply:
Do you mean that we should move all implementations to nuxeo-core ?
I don't have a strong opinion, but I would'nt mind having implementations requiring only nuxeo-core-api in the bulk module
@kevinleturc @efge  ?

Reply:
Agree with @pierre-gautier

Reply:
I think yes action should be defined where they make sense in their context, but for now, it is ok to keep everything in the same module we will see that later in the polish epic.